core/scripts/chaincli/handler/keeper.go

```diff
@@ -90,7 +94,27 @@ func (k *Keeper) DeployKeepers(ctx context.Context) {
 }
 
 // DeployRegistry deploys a new keeper registry.
-func (k *Keeper) DeployRegistry(ctx context.Context) {
+func (k *Keeper) DeployRegistry(ctx context.Context, verify bool) {
+	if verify {
+		if k.cfg.RegistryVersion != keeper.RegistryVersion_2_1 {
+			log.Fatal("keeper registry verification is only supported for version 2.1")
+		}
+		if k.cfg.EtherscanAPIKey == "" || k.cfg.EtherscanAPIKey == "YOUR_ETHERSCAN_KEY" {
+			log.Fatal("please set your etherscan API key in the .env file to verify the registry")
```

Comment:
I like these tests, but can we move them to the config validation function in... `chaincli/config/config.go`? this will make the script fail faster :)

Reply:
These tests are applicable only if the verify flag is on.  In `chaincli/config/config.go` -> Validate(), it validates only the configs, not considering the dynamic flags.

Reply:
ah okay I see